### ISSUE TITLE
twitter_last_read.user.js on mobile.twitter.com

### ISSUE THREAD

**Issue by ghost** [ArmEagle/userscripts#1]
Hi! This script is awesome, could you make it work on mobile.twitter.com too?

**Comment by ArmEagle** [comment]
Hi. Thank you, it indeed makes the official website bearable :).
On my phone I have been going between different Twitter clients. So far I haven't completely liked one yet. But I don't think I'll use the mobile website myself for it. I also don't have much time. But I'll keep this open for when I do see the use for myself.

**Comment by ghost** [comment]
Thank you for answer.
Actually I'm looking for using mobile site for desktop as main site is very heavy on resources after long scrolls.

**Comment by ArmEagle** [comment]
That's a very good point. You may have a good selling point there! :)
I should try the mobile site on my 'slow' laptop/tablet. The normal site is pretty slow regardless of page depth.

**Comment by ghost** [comment]
It seems this script is broken with multiprocess  enabled firefox at least for me. The tweets are marked, the cookie is saved but it doesn't detect them. After closing tab it doesn't show marks anymore.

BTW: When checking script with [tampermonkey addon](https://addons.mozilla.org/en-US/firefox/addon/tampermonkey/) it shows many errors. Maybe one of them is the culprit.

**Comment by ArmEagle** [comment]
Hi peesci. How are you running multi-process Firefox? I just installed Nightly (55.*). I think that has it enabled by default.
I am using GreaseMonkey (as I've always have done) instead of TamperMonkey. I'm not noticing any issues in a short test.

Please report any details in a new issue.

**Comment by ArmEagle** [comment]
And while I'm here anyway; I have looked at the mobile site some time ago. The annoying thing is that there's much less scaffolding (simple classnames) to hook into. It should still be possible though.

**Comment by ghost** [comment]
I think multiprocess isn't enabled by default unless all your addons support it regardless of FF version. For me it's always blocked by addons unless I force-enable it by creating `browser.tabs.remote.force-enable` preference and set it to `true` in FF 52. You can check in about:support for current status just to be sure. As for now I give up enabling it because it gives me to many headaches with many addons. I'll report new issue when I'll be ready to enabling it again.

**Comment by ArmEagle** [comment]
Thanks for reporting though. I can imagine it is more an issue of the extension than the userscript itself though. But we'll probably see in (half) a year or so.

**Comment by ArmEagle** [comment]
Weeks ago I made a start with rewriting the basics from scratch for the mobile site. It's far from complete. But at least there's a start.

I'm now using the mobile website myself on my laptop. Which is up to about half the reads I do. So there's more pressure from myself to fix it too :)

**Comment by ghost** [comment]
That's great thank you! 
I also encountered https://github.com/ArmEagle/userscripts/issues/3 . The third option looks best to deal with it.

**Comment by ArmEagle** [comment]
not working on this script anymore.

I've uploaded what I'm currently working on at https://github.com/ArmEagle/TwitterLastRead .
Still need to cleanup and add/change several things. See todo list in src/main.js.

Closing this issue now.
